tests/post/steps/test_service_configuration.py

```diff
@@ -0,0 +1,239 @@
+    "'{namespace}'"))
+def get_deployments(
+    host,
+    version,
+    k8s_appsv1_client,
+    k8s_client,
+    request,
+    value,
+    path,
+    service,
+    namespace
+
+):
+    def _wait_for_deployment():
+        try:
+            response = k8s_appsv1_client.read_namespaced_deployment(
+                name=service, namespace=namespace
+            ).to_dict()
+        except Exception as exc:
+            pytest.fail(
+                "Unable to read {} Deployment with error: {!s}".format(
+                    service, exc
+                )
+            )
+        assert ast.literal_eval(value) == get_dict_element(response, path), (
+            "Expected value {} for service {}, got {}".format(
+                value, service, get_dict_element(response, path)
+            )
+        )
+    utils.retry(
+        _wait_for_deployment,
+        times=10,
+        wait=5,
+        name="wait for available Deployments"
+    )
+
+
+# }}}
+
+
+# Helpers {{{
+
+
+def read_namespace_config_map(k8s_client, name, namespace):
+    try:
+        response = k8s_client.read_namespaced_config_map(
+            name, namespace
+        )
+    except Exception as exc:
+        pytest.fail(
+            "Unable to read {} ConfigMap  in namespace {} with error: {!s}"
+            .format(name, namespace, exc)
+        )
+    return response
+
+
+def patch_namespace_config_map(k8s_client, name, namespace, patch):
+    try:
+        response = k8s_client.patch_namespaced_config_map(
+            name, namespace, patch
+        )
+    except Exception as exc:
+        pytest.fail(
+            "Unable to patch ConfigMap {} in namespace {} with error {!s}"
+            .format(name, namespace, exc)
+        )
+    return response
+
+
+def csc_load(full_csc, name, namespace):
+    try:
+        csc_obj = yaml.safe_load(full_csc.data['config.yaml'])
+    except yaml.YAMLError as exc:
+        raise Exception(
+            'Invalid YAML format in ConfigMap {} found in namespace {}: {!s}'
+            .format(name, namespace, exc)
+        )
+    except Exception as exc:
+        raise Exception(
+            "Failed loading `config.yaml` from ConfigMap {} in namespace {}: "
+            "{!s}".format(name, namespace, exc)
+        )
+    return csc_obj
+
+
+def get_dict_element(data, path, delimiter='.'):
+    '''
+    Traverse a dict using a 'delimiter' on a target string.
+    getitem(a, b) returns the value of a at index b
+    '''
+    return functools.reduce(operator.getitem, path.split(delimiter), data)
+
+
+def set_dict_element(data, path, new_value, delimiter='.'):
+    '''
+    Traverse a nested dict using a delimiter on a target string
+    replaces the value of a key within a dictionary and returns the new dict
+    '''
+    required_key = path.split(delimiter)[-1]
+    for k, v in data.items():
+        if isinstance(v, dict):
+            set_dict_element(v, path, new_value)
+        if k == required_key:
+            data[k] = new_value
```

Comment:
What ? No !!

Use `.get` or like I suggest you, you can basically use `get_dict_element` with `path.split(delimiter)[:-1]` (with a join or making `get_dict_element` handle list (which is the cleaner solution IMO)

Reply:
Instead of making `get_dict_element` handling both a list and a string (which is not *that* clean) I think it can also be expressed as:

```python
def set_dict_element(data, path, value, delimiter='.'):
    path, _, key = path.rpartition('.')
    (get_dict_element(data, path) if path else data)[key] = value
    return data
```

Reply:
ok